lisa/util/__init__.py

```diff
@@ -352,6 +356,32 @@ def enable(self) -> None:
         self._switch(True)
 
 
+class MutexObject(Generic[T]):
+    """
+    This is a wrapper class which can be used to perform actions of any objects in
+    atomic manner. This is helpful in synchronization when a single object is shared
+    with multiple threads.
+    For example,
+        obj = MutexObject(someObject)
+        with obj.get_object() as mutexobj:
+            mutexobj.method()
+
+    This is same as doing someObject.method() atomically.
+    """
+
+    __object: T
```

Comment:
It doesn't need to specify type in class level. It can be along with in `__init__`.